### PR TITLE
Stakeholder Tool

### PR DESCRIPTION
**Pull request by gwynndp**
## Description

Make Stakeholder tool more user friendly.

**Issue(s) addressed**
- get unlinked stakeholders based on the current stakeholder detail and the logged in account
- format the Add form: add initial state, format fields, close modal on success
- format Filter form: add initial state, format fields
- optimize linking and unlinking state changes
- add state for count

**What kind of change(s) does this PR introduce?**

- [x] Enhancement

**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## Breaking change

**Does this PR introduce a breaking change?**

- [x] No breaking changes

## Other useful information

- This PR builds upon https://github.com/Greenstand/treetracker-admin-client/pull/200
- corresponds with API PR: https://github.com/Greenstand/treetracker-stakeholder-api/pull/4

### CODE REVIEW

File: src/context/AppContext.js [hunk 1]
```diff
@@ -5,9 +5,10 @@ import axios from 'axios';
 import VerifyView from '../views/VerifyView';
 import GrowersView from '../views/GrowersView';
 import CapturesView from '../views/CapturesView';
-import EarningsView from '../views/EarningsView/EarningsView';
-import PaymentsView from '../views/PaymentsView/PaymentsView';
-import MessagingView from 'views/MessagingView';
+// import EarningsView from '../views/EarningsView/EarningsView';
```
Comment (nmcharlton) [inline src/context/AppContext.js:8]: We need to reinstate all of the commented out tools before merging.

File: src/context/AppContext.js [hunk 2]
```diff
@@ -123,6 +125,26 @@ function getRoutes(user) {
...
+      component: Stakeholder,
+      icon: AccountTreeIcon,
+      //TODO this is temporary, need to add stakeholders policy
+      // disabled: false,
+      disabled:
+        process.env.REACT_APP_ENABLE_STAKEHOLDERS !== 'true' ||
+        !hasPermission(user, [
+          POLICIES.SUPER_PERMISSION,
+          POLICIES.APPROVE_TREE,
+          // POLICIES.LIST_STAKEHOLDERS,
```
Comment (nmcharlton) [inline src/context/AppContext.js:140]: New policies have been assigned to the _Stakeholder Manager_ role in the dev, test and prod databases and can be assigned to users.

**Review by nmcharlton — COMMENTED** [review]
I've made a start on the review but have a little way to go.
Looking great so far though!

**Review by nmcharlton — CHANGES_REQUESTED** [review]
Sorry, I added a few comments a while ago but I looks like I didn't submit them.

File: src/api/stakeholders.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
+import { handleResponse, handleError, getOrganizationId } from './apiUtils';
+import { session } from '../models/auth';
+
+const STAKEHOLDER_API = process.env.REACT_APP_STAKEHOLDER_API_ROOT;
```
[the remark this answers is not in the document]
Comment (gwynndp) [reply src/api/stakeholders.js:4]: I've added the current version there now

File: src/components/Stakeholders/Add.js [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+    {
+      name: 'Not set',
+      value: 'Not set', // undefined works
+    },
+    {
+      name: 'Organization',
+      value: 'Organization',
+    },
+    {
+      name: 'Grower',
```
[the remark this answers is not in the document]
Comment (gwynndp) [reply src/components/Stakeholders/Add.js:60]: Good question!  I can change it to Person to match the database types now

**Comment by gwynndp** [comment]
@nmcharlton I've pushed through an update with your requested changes only.  Let's merge this so that I can integrate it with the next PR.

File: .env.development [hunk 5]
```diff
@@ -6,10 +6,22 @@ REACT_APP_QUERY_API_ROOT=https://dev-k8s.treetracker.org/query
 REACT_APP_REPORTING_API_ROOT=https://dev-k8s.treetracker.org/reporting
 REACT_APP_REGION_API_ROOT=https://dev-k8s.treetracker.org/regions
 REACT_APP_TREETRACKER_API_ROOT=https://dev-k8s.treetracker.org/treetracker
+<<<<<<< HEAD
```
Comment (nmcharlton) [inline .env.development:9]: Just need to remove these conflict markers and any duplicate entries.

**Review by nmcharlton — CHANGES_REQUESTED** [review]

**Comment by gwynndp** [comment]
Whoops!  & Thank you!

File: src/api/apiUtils.js [hunk 6]
```diff
@@ -23,6 +23,7 @@ export async function handleResponse(response) {
 
 // we should call an error logging service, but
 export function handleError(error) {
+  console.log('handleError', error);
```
Comment (nmcharlton) [inline src/api/apiUtils.js:26]: If this logging needs to be here, it should use `log.debug()` or similar.

File: src/api/stakeholders.js [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+
+    return fetchJSON(query, options);
+  },
+
+  getUnlinkedStakeholders(id, abortController) {
+    const orgId = getOrganizationId();
+    let query = `${STAKEHOLDER_API}/relations`;
+
+    if (id && orgId && orgId !== id) {
+      query += `/${id}?isRelation=${false}&owner_id=${orgId}`;
```
Comment (nmcharlton) [inline src/api/stakeholders.js:47]: Case in API parameters should be consistent. I'd expect `isRelation` to be `is_relation`.

File: src/context/AppContext.js [hunk 8]
```diff
@@ -149,6 +151,20 @@ function getRoutes(user) {
...
+    {
+      name: 'Stakeholders',
+      linkTo: '/stakeholders',
+      component: Stakeholder,
+      icon: AccountTreeIcon,
+      disabled:
+        process.env.REACT_APP_ENABLE_STAKEHOLDERS !== 'true' ||
+        !hasPermission(user, [
+          POLICIES.SUPER_PERMISSION,
+          POLICIES.APPROVE_TREE,
```
Comment (nmcharlton) [inline src/context/AppContext.js:163]: I don't think `APPROVE_TREE` belongs here.

**Comment by gwynndp** [comment]
@nmcharlton I noticed you weren't on the [other PR](https://github.com/Greenstand/treetracker-admin-client/pull/340), so I added you.

The changes you requested here aren't in the other PR but I'll rebase there once this is merged.

**Pull request closed without merging**